src/pt/readmangas/src/eu/kanade/tachiyomi/extension/pt/readmangas/ReadMangas.kt

```diff
@@ -179,62 +156,60 @@ class ReadMangas() : HttpSource() {
+    private fun chapterListRequest(manga: SManga, page: Int, chapterToken: String): Request {
         val id = manga.url.substringAfterLast("#")
-        val input = buildJsonObject {
-            put(
-                "0",
+        val payload = buildJsonArray {
+            add(
                 buildJsonObject {
-                    put(
-                        "json",
-                        buildJsonObject {
-                            put("id", id)
-                            put("page", page)
-                            put("limit", 50)
-                            put("sort", "desc")
-                            put("search", "")
-                        },
-                    )
+                    put("id", id)
+                    put("page", page)
+                    put("limit", 50)
+                    put("sort", "desc")
+                    put("search", "\$undefined")
                 },
             )
         }
 
-        val url = "$baseUrl/api/deprecated/chapter.publicAllChapters".toHttpUrl().newBuilder()
-            .addQueryParameter("batch", "1")
-            .addQueryParameter("input", input.toString())
+        val newHeaders = headers.newBuilder()
+            .set("Next-Action", chapterToken)
             .build()
 
-        val encodedUrl = URLEncoder.encode(manga.url.substringBeforeLast("#"), "UTF-8")
+        return POST("$baseUrl/title/$id", newHeaders, payload.toRequestBody())
+    }
 
-        val apiHeaders = headers.newBuilder()
-            .set("Referer", "$baseUrl$encodedUrl")
-            .set("Content-Type", "application/json")
-            .set("Cache-Control", "no-cache")
-            .build()
+    private fun findChapterToken(manga: SManga): String {
+        var response = client.newCall(super.chapterListRequest(manga)).execute()
+        val document = response.asJsoup()
+
+        val scriptUlr = document.selectFirst("""script[src*="%5Boid%5D/page"]""")
+            ?.absUrl("src")
+            ?: throw Exception("Token não encontrado")
 
-        return GET(url, apiHeaders)
+        response = client.newCall(GET(scriptUlr, headers)).execute()
+
+        return TOKEN_REGEX.find(response.body.string())?.groups?.get(1)?.value
+            ?: throw Exception("Não foi possivel obter token")
     }
 
     override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapterToken = findChapterToken(manga)
         val chapters = mutableListOf<SChapter>()
         var page = 1
         do {
-            val response = tryFetchChapterPage(manga, page++)
-            val dto = response
-                .parseAs<List<WrapperResult<ChapterListDto>>>()
-                .firstNotNullOf { it.result }
-                .data.json
+            val response = tryFetchChapterPage(manga, page++, chapterToken)
+            val json = CHAPTERS_REGEX.find(response.body.string())?.groups?.get(0)?.value
+            val dto = json!!.parseAs<ChapterListDto>()
```

Comment:
Will this (non-null assert) mean a failure on page 4 mean chapters from page 1-3 are discarded?